pony/converting.py

```diff
@@ -1,249 +1,247 @@
-
-    d = match.groupdict()
-    year, day, month = d['year'], d['day'], d.get('month')
-
-    if month is None:
-        for key, value in month_dict.items():
-            if key in s: month = value; break
-        else: raise ValueError('Unrecognized datetime format')
-
-    hh, mm, ss, mcs = _extract_time_parts(d)
-    return datetime(int(year), int(month), int(day), hh, mm, ss, mcs)
-
-def _extract_time_parts(groupdict):
-    hh, mm, ss, am, pm = map(groupdict.get, ('hh', 'mm', 'ss', 'am', 'pm'))
-
-    if hh is None: hh, mm, ss = 12, 00, 00
-    elif am and hh == '12': hh = 0
-    elif pm and hh != '12': hh = int(hh) + 12
-
-    if isinstance(ss, str) and '.' in ss:
-        ss, mcs = ss.split('.', 1)
-        if len('mcs') < 6: mcs = (mcs + '000000')[:6]
-    else: mcs = 0
-
-    return int(hh), int(mm or 0), int(ss or 0), int(mcs)
-
-def str2timedelta(s):
-    negative = s.startswith('-')
-    if '.' in s:
-        s, fractional = s.split('.')
-        microseconds = int((fractional + '000000')[:6])
-    else: microseconds = 0
-    h, m, s = map(int, s.split(':'))
-    td = timedelta(hours=abs(h), minutes=m, seconds=s, microseconds=microseconds)
-    return -td if negative else td
-
-def timedelta2str(td):
-    total_seconds = td.days * (24 * 60 * 60) + td.seconds
-    microseconds = td.microseconds
-    if td.days < 0:
-        total_seconds = abs(total_seconds)
-        if microseconds:
-            total_seconds -= 1
-            microseconds = 1000000 - microseconds
-    minutes, seconds = divmod(total_seconds, 60)
-    hours, minutes = divmod(minutes, 60)
-    if microseconds: result = '%d:%d:%d.%06d' % (hours, minutes, seconds, microseconds)
-    else: result = '%d:%d:%d' % (hours, minutes, seconds)
-    if td.days >= 0: return result
-    return '-' + result
-
-converters = {
-    int:  (int, str, 'Incorrect number'),
-    float: (float, str, 'Must be a real number'),
-    'IP': (check_ip, str, 'Incorrect IP address'),
-    'positive': (check_positive, str, 'Must be a positive number'),
-    'identifier': (check_identifier, str, 'Incorrect identifier'),
-    'ISBN': (check_isbn, str, 'Incorrect ISBN'),
-    'email': (check_email, str, 'Incorrect e-mail address'),
-    'rfc2822_email': (check_rfc2822_email, str, 'Must be correct e-mail address'),
-    date: (str2date, str, 'Must be correct date (mm/dd/yyyy or dd.mm.yyyy)'),
-    time: (str2time, str, 'Must be correct time (hh:mm or hh:mm:ss)'),
-    datetime: (str2datetime, str, 'Must be correct date & time'),
-    }
-
-def str2py(value, type):
-    if type is None or not isinstance(value, str): return value
-    if isinstance(type, tuple): str2py, py2str, err_msg = type
-    else: str2py, py2str, err_msg = converters.get(type, (type, str, None))
-    try: return str2py(value)
-    except ValidationError: raise
-    except:
-        if value == '': return None
-        raise ValidationError(err_msg or 'Incorrect data')
+from __future__ import absolute_import, print_function
```

Comment:
Hiding whitespace while reviewing this is very helpful, github treats encoding changes as whitespace.